torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +41,244 @@ def decorator(func):
     return decorator
 
 
-@implements([torch.ops.aten.detach.default, torch.ops.aten.detach])
+@implements([torch.ops.aten.detach])
 def noop_detach(func, *args, **kwargs):
     return args[0][0]
 
 
+@implements(
+    [
+        aten.detach.default,
+    ]
+)
+def nf4_detach(aten_op, args, kwargs=None):
+    # nn.Parameter need detach
+    quantized_scalers = aten_op(args[0].quantized_scalers, *args[1:], **kwargs)
+    quantization_factor = aten_op(args[0].quantization_factor, *args[1:], **kwargs)
+    quantized_data = aten_op(args[0].quantized_data, *args[1:], **kwargs)
+    scaler_mean = aten_op(args[0].scaler_mean, *args[1:], **kwargs)
+    nf4 = aten_op(args[0].nf4, *args[1:], **kwargs)
+    tensor_meta = SubclassTensorArgs(
+        args[0].size(),
+        args[0].stride(),
+        args[0].storage_offset(),
+        args[0].dtype,
+        args[0].device,
+        args[0].requires_grad,
+    )
+    return NF4Tensor(
+        tensor_meta,
+        args[0].block_size,
+        args[0].n_blocks,
+        args[0].scaler_block_size,
+        quantized_scalers,
+        quantization_factor,
+        scaler_mean,
+        quantized_data,
+        nf4,
+    )
+
+@implements(
+    [
+        aten.split.Tensor,
+    ]
+)
+def nf4_split(aten_op, args, kwargs=None):
+    if len(args) == 3 and args[2] != 0:
+        raise NotImplementedError(f"aten.split(NF4Tensor, dim={args[2]})")
+    outer_wrapper = args[0]
+    num_chunks = outer_wrapper.size(0) // args[1]
+
+    assert outer_wrapper.quantized_scalers.numel() % num_chunks == 0, f"NF4Tensor.quantized_scalers.numel() not divisible by {num_chunks}"
```

Comment:
Can you reuse the class constructor here for validation? Or maybe a factory function that can construct an NF4 Tensor from components.

Reply:
I kinda cheated before and higjacked tensor_flatten + unflatten to do this for me

Reply:
@drisspg - Yeah I wonder if you think that's a good pattern in general actually (it seems like it). We could use it for a default module reparam implementation and such as well.

Reply:
Yeah I am not sure, in theory it should be the source of truth on at least 1 way to "unpack" and "repack" your tensor subclass